app/src/main/java/com/infomaniak/swisstransfer/ui/components/FileTile.kt

```diff
@@ -0,0 +1,250 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.core.net.toUri
+import coil.compose.AsyncImage
+import coil.request.ImageRequest
+import com.infomaniak.library.filetypes.FileType
+import com.infomaniak.swisstransfer.R
+import com.infomaniak.swisstransfer.ui.images.AppImages
+import com.infomaniak.swisstransfer.ui.images.icons.CrossThick
+import com.infomaniak.swisstransfer.ui.theme.LocalIsDarkMode
+import com.infomaniak.swisstransfer.ui.theme.Margin
+import com.infomaniak.swisstransfer.ui.theme.Shapes
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+import com.infomaniak.swisstransfer.ui.utils.PreviewLargeWindow
+import com.infomaniak.swisstransfer.ui.utils.PreviewSmallWindow
+import com.infomaniak.swisstransfer.ui.utils.fileType
+import com.infomaniak.swisstransfer.ui.utils.hasPreview
+
+// TODO: Get the interface from the shared kmp code
+interface FileUiItem {
+    val uid: String
+    val fileName: String
+    val fileSizeInBytes: Long
+    val mimeType: String?
+    val uri: String
+}
+
+@Composable
+fun FileTile(
+    file: FileUiItem,
+    isRemoveButtonVisible: Boolean,
+    isCheckboxVisible: Boolean,
+    isChecked: () -> Boolean = { false },
+    onClick: () -> Unit,
+    onRemove: (() -> Unit)? = null,
+) {
+    FileTileContent(
+        content = {
+            var displayPreview by rememberSaveable { mutableStateOf(file.hasPreview) }
+
+            if (displayPreview) {
+                FileThumbnail(file.uri.toUri(), onError = { displayPreview = false })
+            } else {
+                FileIcon(file.fileType)
+            }
+        },
+        onClick = onClick,
+        isCheckboxVisible = isCheckboxVisible,
+        isChecked = isChecked,
+        isRemoveButtonVisible = isRemoveButtonVisible,
+        onRemove = onRemove,
+        title = file.fileName,
+        description = Formatter.formatShortFileSize(LocalContext.current, file.fileSizeInBytes),
+    )
+}
+
+@Composable
+private fun FileTileContent(
+    content: @Composable () -> Unit,
+    onClick: () -> Unit,
+    isCheckboxVisible: Boolean,
+    isChecked: () -> Boolean,
+    isRemoveButtonVisible: Boolean,
+    onRemove: (() -> Unit)?,
+    title: String,
+    description: String
+) {
+    Card(
+        onClick = onClick,
+        modifier = Modifier.aspectRatio(164 / 152f),
+        colors = CardDefaults.cardColors(containerColor = SwissTransferTheme.materialColors.background),
+        shape = Shapes.small,
+        border = BorderStroke(width = 1.dp, SwissTransferTheme.materialColors.outlineVariant)
+    ) {
+        Box(
+            modifier = Modifier
+                .weight(1f)
```

Comment:
Check if `weight` is really needed ?

Reply:
Yeah no, it's actually needed we didn't look carefully but this Box is one of the two childs of the Card's ColumnScope. Yeah it's needed